lib/topologies/read_preference.js

```diff
@@ -54,13 +55,54 @@ var ReadPreference = function(preference, tags, options) {
   }
 };
 
+// Support the deprecated `preference` property introduced in the porcelain layer
+Object.defineProperty(ReadPreference.prototype, 'preference', {
+  enumerable: true,
+  get: function() {
+    return this.mode;
+  }
+});
+
+/**
+ * Validate if a mode is legal
+ *
+ * @method
+ * @param {string} mode The string representing the read preference mode.
+ * @return {boolean}
+ */
+ReadPreference.isValid = function(mode) {
```

Comment:
Can we use a `Set` here?

Reply:
like:
```
const validModes = [
  ReadPreference.PRIMARY, 
  ReadPreference.PRIMARY_PREFERRED,
  ReadPreference.SECONDARY,
  ReadPreference.SECONDARY_PREFERRED,
  ReadPreference.NEAREST,
  true,
  false,
  null
];

ReadPreference.isValid = function(mode) {
  return validModes.has(mode);
}
```
?

Reply:
Yeah, either by using the native `Set` (do we get that in Node 4?), or by just doing `validModes.indexOf(x) >=0`

Reply:
okay I'm into it